vault/resource_aws_secret_backend.go

```diff
@@ -99,38 +99,65 @@ func awsSecretBackendResource() *schema.Resource {
 	}
 }
 
-func mountMigrationCustomizeDiff(ctx context.Context, diff *schema.ResourceDiff, meta interface{}) error {
-	if !diff.HasChange("path") {
+func mountMigrationCustomizeDiffFieldPath(ctx context.Context, diff *schema.ResourceDiff, meta interface{}) error {
+	return mountMigrationHelper(ctx, diff, meta, consts.FieldPath)
+}
+
+func mountMigrationHelper(_ context.Context, diff *schema.ResourceDiff, meta interface{}, mountField string) error {
+	if !diff.HasChange(mountField) {
 		return nil
 	}
 
-	o, _ := diff.GetChange("path")
+	o, _ := diff.GetChange(mountField)
 	if o == "" {
 		return nil
 	}
 
 	// Mount Migration is only available for versions >= 1.10
-	client, e := provider.GetClient(diff, meta)
-	if e != nil {
-		return e
-	}
-
-	remountEnabled, _, err := semver.GreaterThanOrEqual(ctx, client, consts.VaultVersion10)
+	minVersion, err := version.NewVersion(consts.VaultVersion10)
 	if err != nil {
 		return err
 	}
+	remountSupported := provider.IsAPISupported(meta, minVersion)
 
-	if !remountEnabled {
+	if !remountSupported {
 		// Mount migration not available
 		// Destroy and recreate resource
-		if err := diff.ForceNew("path"); err != nil {
+		if err := diff.ForceNew(mountField); err != nil {
 			return err
 		}
 	}
 
 	return nil
 }
 
+func remountToNewPath(d *schema.ResourceData, client *api.Client, mountField string, isAuthMount bool) (string, error) {
```

Comment:
This seems like a general purpose function, suggest moving it to the `util` package.